### PR TITLE
[refactor][cli][PIP-280] Create new `pulsar-cli-utils` module 

### PR DESCRIPTION
**Pull request by JooHyukKim**
PIP: #20691 

### Motivation

As PIP-280 #20691 may span cross module, we need a module to provide CLI related (though mostly `JCommander` atm) isolation. Note that `pulsar-common` was not considered because it's common (too broad).

### Modifications

- Add new module `pulsar-cli-utils`
- Implement `Converter`s for meaurement unit (time,byte) conversion
- Implement `Validator`s for

### Note

- This PR blocks #20764

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [x] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/JooHyukKim/pulsar/pull/19

### CODE REVIEW

**Comment by tisonkun** [comment]
@JooHyukKim you can now rebase this branch on the fixed master :D

File: pulsar-cli-utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        <artifactId>lightproto-maven-plugin</artifactId>
+        <version>${lightproto-maven-plugin.version}</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>generate</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
```
Comment (BewareMyPower) [inline pulsar-cli-utils/pom.xml:88]: Why do you need this plugin? I see no protobuf files needed in this PR

**Comment by tisonkun** [comment]
I'm thinking if we would include this patch in 3.1.0 since the code freeze will happen the next week.

This seems part of PIP-280 and the whole refactor of PIP-280 can include other tasks. If we want to include PIP-280 in 3.1.0, maybe more reviewers should involve. Otherwise, we postpone the proposal to the next cycle so we don't even merge this first step in 3.1.0.

cc @Technoboy-

File: pulsar-cli-utils/src/main/java/org/apache/pulsar/cli/converters/RelativeTimeUtil.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * Mirror of {@link org.apache.pulsar.common.util.RelativeTimeUtil}.
+ */
+@UtilityClass
+public class RelativeTimeUtil {
+    public static long parseRelativeTimeInSeconds(String relativeTime) {
+        if (relativeTime.isEmpty()) {
+            throw new IllegalArgumentException("time cannot be empty");
+        }
+
+        int lastIndex =  relativeTime.length() - 1;
```
Comment (gaoran10) [inline pulsar-cli-utils/src/main/java/org/apache/pulsar/cli/converters/RelativeTimeUtil.java:36]:
```suggestion
        int lastIndex = relativeTime.length() - 1;
```

**Review by gaoran10 — APPROVED** [review]
Great work! LGTM

File: pulsar-cli-utils/src/main/java/org/apache/pulsar/cli/converters/RelativeTimeUtil.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            throw new IllegalArgumentException("Invalid time unit '" + lastChar + "'");
+        }
+    }
+
+    /**
+     * Convert nanoseconds to seconds and keep three decimal places.
+     * @param ns
+     * @return seconds
+     */
+    public static double nsToSeconds(long ns) {
```
Comment (BewareMyPower) [inline pulsar-cli-utils/src/main/java/org/apache/pulsar/cli/converters/RelativeTimeUtil.java:74]: This method is not tested.

**Review by BewareMyPower — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your contribution!

**Comment by tisonkun** [comment]
@JooHyukKim Maybe you can open a tracking issue to list out the subtasks and finish them one by one. In this way, we can share the same page what is remaining and maybe offer some helps.

**Pull request merged**